src/libopensc/card-tcos.c

```diff
@@ -124,88 +124,88 @@ static int tcos_construct_fci(const sc_file_t *file,
-                buf[n++] = file->record_length;
-        }
+	if ( (file->ef_structure & 7) > 1) {
+		/* record structured file */
+		buf[n++] = 0x41; /* indicate 3rd byte */
+		buf[n++] = file->record_length;
+	}
 	sc_asn1_put_tag(0x82, buf, n, p, 8, &p);
 
-        /* File identifier */
+	/* File identifier */
 	buf[0] = (file->id >> 8) & 0xFF;
 	buf[1] = file->id & 0xFF;
 	sc_asn1_put_tag(0x83, buf, 2, p, 16, &p);
 
-        /* Directory name */
-        if (file->type == SC_FILE_TYPE_DF) {
-                if (file->namelen) {
-                        sc_asn1_put_tag(0x84, file->name, file->namelen,
-                                        p, 16, &p);
-                }
-                else {
-                        /* TCOS needs one, so we use a faked one */
-                        snprintf ((char *) buf, sizeof(buf)-1, "foo-%lu",
-                                  (unsigned long) time (NULL));
-                        sc_asn1_put_tag(0x84, buf, strlen ((char *) buf), p, 16, &p);
-                }
-        }
-
-        /* File descriptor extension */
-        if (file->prop_attr_len && file->prop_attr) {
+	/* Directory name */
+	if (file->type == SC_FILE_TYPE_DF) {
+		if (file->namelen) {
+			sc_asn1_put_tag(0x84, file->name, file->namelen,
+					p, 16, &p);
+		}
+		else {
+			/* TCOS needs one, so we use a faked one */
+			snprintf ((char *) buf, sizeof(buf)-1, "foo-%lu",
+					(unsigned long) time (NULL));
+			sc_asn1_put_tag(0x84, buf, strlen ((char *) buf), p, 16, &p);
+		}
+	}
+
+	/* File descriptor extension */
+	if (file->prop_attr_len && file->prop_attr) {
 		n = file->prop_attr_len;
 		memcpy(buf, file->prop_attr, n);
-        }
-        else {
-                n = 0;
-                buf[n++] = 0x01; /* not invalidated, permanent */
-                if (file->type == SC_FILE_TYPE_WORKING_EF) 
-                        buf[n++] = 0x00; /* generic data file */
-        }
-        sc_asn1_put_tag(0x85, buf, n, p, 16, &p);
-
-        /* Security attributes */
+	}
+	else {
+		n = 0;
+		buf[n++] = 0x01; /* not invalidated, permanent */
+		if (file->type == SC_FILE_TYPE_WORKING_EF)
+			buf[n++] = 0x00; /* generic data file */
+	}
+	sc_asn1_put_tag(0x85, buf, n, p, 16, &p);
+
+	/* Security attributes */
 	if (file->sec_attr_len && file->sec_attr) {
 		memcpy(buf, file->sec_attr, file->sec_attr_len);
 		n = file->sec_attr_len;
 	}
-        else {
-                /* no attributes given - fall back to default one */
-                memcpy (buf+ 0, "\xa4\x00\x00\x00\xff\xff", 6); /* select */
-                memcpy (buf+ 6, "\xb0\x00\x00\x00\xff\xff", 6); /* read bin */
-                memcpy (buf+12, "\xd6\x00\x00\x00\xff\xff", 6); /* upd bin */
-                memcpy (buf+18, "\x60\x00\x00\x00\xff\xff", 6); /* admin grp*/
-                n = 24;
-        }
-        sc_asn1_put_tag(0x86, buf, n, p, sizeof (buf), &p);
-
-        
-        /* fixup length of FCI */
-        out[1] = p - out - 2;
+	else {
```

Comment:
I tried to find the last discussion about coding style, but I was not successful. The only thing I found was some ancient wiki referencing linux kernel coding style, which allows only `} else {` on single line so I would try to stick with that as we rewrite things.

https://github.com/OpenSC/OpenSC/wiki/Development-Policy